Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionSidebarCocoa.mm

```diff
@@ -0,0 +1,299 @@
+}
+
+template<typename T>
+static std::optional<T *> nilToNullopt(T *maybeNil)
+{
+    if (maybeNil)
+        return maybeNil;
+    return std::nullopt;
+}
+
+WebExtensionSidebar::WebExtensionSidebar(WebExtensionContext& context, IsDefaultSidebar isDefault) : WebExtensionSidebar(context, std::nullopt, std::nullopt, isDefault) { };
+
+WebExtensionSidebar::WebExtensionSidebar(WebExtensionContext& context, WebExtensionTab& tab) : WebExtensionSidebar(context, tab, std::nullopt, IsDefaultSidebar::No) { };
+
+WebExtensionSidebar::WebExtensionSidebar(WebExtensionContext& context, WebExtensionWindow& window) : WebExtensionSidebar(context, std::nullopt, window, IsDefaultSidebar::No) { };
+
+WebExtensionSidebar::WebExtensionSidebar(WebExtensionContext& context, std::optional<Ref<WebExtensionTab>> tab, std::optional<Ref<WebExtensionWindow>> window, IsDefaultSidebar isDefault)
+    // for this API, we never want to override both window and tab -- can only do one at a time
+    // if we somehow (erroneously) get both, prefer tab (most specific)
+    : m_context(WeakRef(context)), m_tab(tab), m_window(window && tab ? std::nullopt : window), m_isDefault(isDefault)
+{
+
+    // if this is the default action, initialize with default sidebar path / title if present
+    if (isDefaultSidebar()) {
+        auto& extension = context.extension();
+        m_titleOverride = getDefaultSidebarTitleFromExtension(extension);
+        m_sidebarPathOverride = getDefaultSidebarPathFromExtension(extension);
+        m_iconsOverride = getDefaultIconsDictFromExtension(extension);
+    }
+}
+
+std::optional<Ref<WebExtensionContext const>> WebExtensionSidebar::extensionContext() const
+{
+    if (m_context.ptr())
+        return m_context.get();
+    return std::nullopt;
+}
+
+std::optional<Ref<WebExtensionContext>> WebExtensionSidebar::extensionContext()
+{
+    return const_cast<WebExtensionSidebar const*>(this)->extensionContext().transform([](auto ctx) -> Ref<WebExtensionContext> {
+        return const_cast<WebExtensionContext&>(ctx.get());
+    });
+}
+
+std::optional<Ref<WebExtensionTab const>> WebExtensionSidebar::tab() const
+{
+    return m_tab.transform([](auto tab) -> Ref<WebExtensionTab const> {
+        return tab.get();
+    });
+}
+
+std::optional<Ref<WebExtensionTab>> WebExtensionSidebar::tab()
+{
+    return const_cast<WebExtensionSidebar const*>(this)->tab().transform([](auto tab) -> Ref<WebExtensionTab> {
+        return const_cast<WebExtensionTab&>(tab.get());
+    });
+}
+
+std::optional<Ref<WebExtensionWindow const>> WebExtensionSidebar::window() const
+{
+    return m_window.transform([](auto window) -> Ref<WebExtensionWindow const> {
+        return window.get();
+    });
+}
+
+std::optional<Ref<WebExtensionWindow>> WebExtensionSidebar::window()
+{
+    return const_cast<WebExtensionSidebar const*>(this)->window().transform([](auto window) -> Ref<WebExtensionWindow> {
+        return const_cast<WebExtensionWindow&>(window.get());
+    });
+}
```

Comment:
Should just have one `window()` and one `tab()` that is `std::optional<Ref<T>>`, no `const`.

`transform` and lambdas seem like overkill for these simple getters.

Reply:
Understood, I can definitely remove the `const`/non-`const` duplication for these